plugins/gatsby-source-data/main.js

```diff
@@ -140,31 +139,40 @@ const getPlaylistsFromYoutube = async (orgData, apiKey) => {
           const playlistObj = {
```

Comment:
Few comments here:
1. reduce nesting
2. use functional programming for better readability
3. Parallel API calls with `Promise.all()`

Lets fix no. 1 at least with earlier returns

```
for (org in orgData) {
  if (!org.youtubeId) {
    return
  }

  const channelPlaylists = await yt.getChannelPlaylists(org.youtubeId);
  const playlistArr = [];
  for (playlist of channelPlaylists) {
    const tags = getValidTags(playlist);
    if(tags.category.length <= 0) {
        // Only process the playlist if it has valid tags.
        return
    }

    ...
}
```

I have refactored the first few lines to explain my point, but if there are any place below we can apply this to reduce the nesting, please do.

Reply:
Should it be continue instead of return?

Reply:
yes it should be `continue`, updated the code example above.